src/gluonts/ext/r_forecast/_univariate_predictor.py

```diff
@@ -95,79 +99,59 @@ def __init__(
             ]
+            params["intervals"] = sorted(
+                set([level for level, _ in intervals_info])
+            )
 
-    def get_levels(self, quantiles: List[float]):  ## code_diff
-        percentage_levels = [2 * abs(0.5 - quantile) for quantile in quantiles]
-        levels = [
-            round(percentage_level * 100)
-            for percentage_level in percentage_levels
-        ]
-        return sorted(zip(levels, quantiles))
+        self.params.update(params)
 
-    def _get_r_forecast(self, data: Dict, params: Dict) -> Dict:
+    def _get_r_forecast(self, data: Dict) -> Dict:
         make_ts = self._stats_pkg.ts
-        r_params = self._robjects.vectors.ListVector(params)
+        r_params = self._robjects.vectors.ListVector(self.params)
         vec = self._robjects.FloatVector(data["target"])
         ts = make_ts(vec, frequency=self.period)
         forecast = self._r_method(ts, r_params)
 
-        forecast_dict = dict(
-            zip(forecast.names, map(self._unlist, list(forecast)))
-        )
-
-        if "quantiles" in forecast_dict or "upper_quantiles" in forecast_dict:
-
-            def from_interval_to_level(interval: int, side: str):
-                if side == "upper":
-                    level = 50 + interval / 2
-                elif side == "lower":
-                    level = 50 - interval / 2
-                else:
-                    raise ValueError
-                return level / 100
+        forecast_dict = dict(zip(forecast.names, map(unlist, list(forecast))))
 
-            # Post-processing quantiles on then Python side for the
-            # convenience of asserting and debugging.
+        if "quantiles" in self.params["output_types"]:
             upper_quantiles = [
-                str(from_interval_to_level(interval, side="upper"))
-                for interval in params["intervals"]
+                str(interval_to_quantile_level(interval, side="upper"))
+                for interval in self.params["intervals"]
             ]
 
             lower_quantiles = [
-                str(from_interval_to_level(interval, side="lower"))
-                for interval in params["intervals"]
+                str(interval_to_quantile_level(interval, side="lower"))
+                for interval in self.params["intervals"]
             ]
 
-            # Median forecasts would be available at two places:
-            # Lower 0 and Higher 0 (0-prediction interval)
             forecast_dict["quantiles"] = dict(
                 zip(
-                    lower_quantiles + upper_quantiles[1:],
+                    lower_quantiles + upper_quantiles,
                     forecast_dict["lower_quantiles"]
-                    + forecast_dict["upper_quantiles"][1:],
+                    + forecast_dict["upper_quantiles"],
                 )
             )
 
-            # `QuantileForecast` allows "mean" as the key;
-            # we store them as well since they can differ from median.
```

Comment:
The reason I had this here is to avoid returning a wrong value without warning: the user sends only `"quantiles"` in `output_types` and obtains a `QuantileForecast` which in this case returns `p50` when requested for `mean`. So if the user wants to evaluate this forecast using a point metric and requests for `mean` (because it supports mean), then `QuantileForecast` would return a wrong value without raising any warning. 

So it is better to always include mean while creating `QuantileForecast`.

Reply:
Ok now I see what you mean